File: Examples.VB/Features/PDFExporting/ExportChart/PieChartPdf.vb
Namespace Features.PDFExporting.ExportChart
    Public Class PieChartPdf
        Inherits ExampleBase

        Public Overrides Sub Execute(workbook As Excel.Workbook)
            Dim worksheet As IWorksheet = workbook.Worksheets(0)
            Dim shape As IShape = worksheet.Shapes.AddChart(ChartType.Pie, 20, 20, 360, 230)
            worksheet.Range("A20:B23").Value = New Object(,) {
                {"Blue", 1}, {"Red", 2}, {"Green", 3}, {"Purple", 4}
            }

            With shape.Chart
                .SeriesCollection.Add(worksheet.Range("A20:B23"), RowCol.Columns)
                .ChartTitle.Text = "Pie Chart"
                .Legend.Position = LegendPosition.Right
            End With
        End Sub

        Public Overrides ReadOnly Property IsNew As Boolean
            Get
                Return True
            End Get
        End Property

        Public Overrides ReadOnly Property SavePdf As Boolean
            Get
                Return True
            End Get
        End Property

        Public Overrides ReadOnly Property ShowViewer As Boolean
            Get
                Return False
            End Get
        End Property
    End Class
End Namespace

File: Examples/Features/PDFExporting/ExportChart/PieChartPdf.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace GrapeCity.Documents.Excel.Examples.Features.PDFExporting.ExportChart
{
    public class PieChartPdf : ExampleBase
    {
        public override void Execute(GrapeCity.Documents.Excel.Workbook workbook)
        {
            IWorksheet worksheet = workbook.Worksheets[0];
            GrapeCity.Documents.Excel.Drawing.IShape shape = worksheet.Shapes.AddChart(GrapeCity.Documents.Excel.Drawing.ChartType.Pie, 20, 20, 360, 230);
            worksheet.Range["A20:B23"].Value = new object[,] {
                { "Blue", 1 },
                { "Red", 2 },
                { "Green", 3 },
                { "Purple", 4 },
            };
            shape.Chart.SeriesCollection.Add(worksheet.Range["A20:B23"], GrapeCity.Documents.Excel.Drawing.RowCol.Columns);
            shape.Chart.ChartTitle.Text = "Pie Chart";
            shape.Chart.Legend.Position = GrapeCity.Documents.Excel.Drawing.LegendPosition.Right;
        }

        public override bool SavePdf
        {
            get
            {
                return true;
            }
        }

        public override bool ShowViewer
        {
            get
            {
                return false;
            }
        }
    }
}
